diff --git a/libraries/Network/src/NetworkClient.cpp b/libraries/Network/src/NetworkClient.cpp
--- a/libraries/Network/src/NetworkClient.cpp
+++ b/libraries/Network/src/NetworkClient.cpp
@@ -217,7 +217,7 @@ int NetworkClient::connect(IPAddress ip, uint16_t port, int32_t timeout_ms) {
     struct sockaddr_in6 *tmpaddr = (struct sockaddr_in6 *)&serveraddr;
     sockfd = socket(AF_INET6, SOCK_STREAM, 0);
     tmpaddr->sin6_family = AF_INET6;
-    memcpy(tmpaddr->sin6_addr.un.u8_addr, &ip[0], 16);
+    memcpy(tmpaddr->sin6_addr.s6_addr, &ip[0], sizeof(tmpaddr->sin6_addr.s6_addr));
     tmpaddr->sin6_port = htons(port);
     tmpaddr->sin6_scope_id = ip.zone();
   } else {
@@ -401,5 +401,5 @@ size_t NetworkClient::write(const uint8_t *buf, size_t size) {
     FD_ZERO(&set);                       // empties the set
     FD_SET(socketFileDescriptor, &set);  // adds FD to the set
-    tv.tv_sec = 0;
-    tv.tv_usec = WIFI_CLIENT_SELECT_TIMEOUT_US;
+    tv.tv_sec = WIFI_CLIENT_SELECT_TIMEOUT_US / 1000000;
+    tv.tv_usec = WIFI_CLIENT_SELECT_TIMEOUT_US % 1000000;
     retry--;
@@ -514,5 +514,6 @@ size_t NetworkClient::readBytes(char *buffer, size_t length) {
   size_t left = length, sofar = 0;
-  int r = 0, to = millis() + getTimeout();
+  int r = 0;
+  unsigned long to = millis() + getTimeout();
   while (left) {
     r = read((uint8_t *)buffer + sofar, left);
     if (r < 0) {
@@ -617,7 +618,7 @@ IPAddress NetworkClient::remoteIP(int fd) const {
   // IPv6, but it might be IPv4 mapped address
   if (((struct sockaddr *)&addr)->sa_family == AF_INET6) {
     struct sockaddr_in6 *saddr6 = (struct sockaddr_in6 *)&addr;
-    if (IN6_IS_ADDR_V4MAPPED(saddr6->sin6_addr.un.u32_addr)) {
+    if (IN6_IS_ADDR_V4MAPPED(&saddr6->sin6_addr)) {
       return IPAddress(IPv4, (uint8_t *)saddr6->sin6_addr.s6_addr + IPADDRESS_V4_BYTES_INDEX);
     } else {
       return IPAddress(IPv6, (uint8_t *)(saddr6->sin6_addr.s6_addr), saddr6->sin6_scope_id);
@@ -659,7 +660,7 @@ IPAddress NetworkClient::localIP(int fd) const {
   // IPv6, but it might be IPv4 mapped address
   if (((struct sockaddr *)&addr)->sa_family == AF_INET6) {
     struct sockaddr_in6 *saddr6 = (struct sockaddr_in6 *)&addr;
-    if (IN6_IS_ADDR_V4MAPPED(saddr6->sin6_addr.un.u32_addr)) {
+    if (IN6_IS_ADDR_V4MAPPED(&saddr6->sin6_addr)) {
       return IPAddress(IPv4, (uint8_t *)saddr6->sin6_addr.s6_addr + IPADDRESS_V4_BYTES_INDEX);
     } else {
       return IPAddress(IPv6, (uint8_t *)(saddr6->sin6_addr.s6_addr), saddr6->sin6_scope_id);
diff --git a/libraries/Network/src/NetworkEvents.h b/libraries/Network/src/NetworkEvents.h
--- a/libraries/Network/src/NetworkEvents.h
+++ b/libraries/Network/src/NetworkEvents.h
@@ -8,7 +8,9 @@
 #include "sdkconfig.h"
 #include "soc/soc_caps.h"
+#include "esp_bit_defs.h"
 #include "esp_err.h"
 #include "esp_event.h"
+#include "esp_idf_version.h"
 #include "esp_netif_types.h"
 #if CONFIG_ETH_ENABLED
 #include "esp_eth_driver.h"
diff --git a/libraries/Network/src/NetworkUdp.cpp b/libraries/Network/src/NetworkUdp.cpp
--- a/libraries/Network/src/NetworkUdp.cpp
+++ b/libraries/Network/src/NetworkUdp.cpp
@@ -308,7 +308,7 @@ int NetworkUDP::parsePacket() {
   struct sockaddr_storage si_other_storage;  // enough storage for v4 and v6
   socklen_t slen = sizeof(sockaddr_storage);
   int len = 0;
-  if (ioctl(udp_server, FIONREAD, &len) == -1) {
+  if (lwip_ioctl(udp_server, FIONREAD, &len) == -1) {
     log_e("could not check for data in buffer length: %d", errno);
     return 0;
   }
